homeassistant/components/vicare/sensor.py

```diff
@@ -222,6 +234,92 @@ class ViCareSensorEntityDescription(SensorEntityDescription, ViCareRequiredKeysM
         device_class=SensorDeviceClass.ENERGY,
         state_class=SensorStateClass.TOTAL_INCREASING,
     ),
+    ViCareSensorEntityDescription(
+        key=SENSOR_SOLAR_COLLECTOR_TEMPERATURE,
+        name="Solar Collector Temperature",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        value_getter=lambda api: api.getSolarCollectorTemperature(),
+        device_class=SensorDeviceClass.TEMPERATURE,
+    ),
+    ViCareSensorEntityDescription(
+        key=SENSOR_SOLAR_HOTWATER_TEMPERATURE,
+        name="Solar Hotwater Temperature",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        value_getter=lambda api: api.getSolarStorageTemperature(),
+        device_class=SensorDeviceClass.TEMPERATURE,
+    ),
+    ViCareSensorEntityDescription(
+        key=SENSOR_SOLAR_POWER_PRODUCTION_TODAY,
+        name="Solar power production today",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        value_getter=lambda api: api.getSolarPowerProductionToday(),
+        unit_getter=lambda api: api.getSolarPowerProductionUnit(),
```

Comment:
`unit_getter` seems not to be part of the mixin dataclass?

Reply:
merging latest branch 'home-assistant:dev' into my patch-3 fixed that issue.
now `unit_getter` should be part of the mixin dataclass.